core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -900,8 +912,6 @@ class LogTest extends JUnitSuite {
     // files should be renamed
     val segments = log.logSegments.toArray
     val oldFiles = segments.map(_.log.file) ++ segments.map(_.index.file)
-    // expire all segments
```

Comment:
Perhaps we should set LogConfig.RetentionMsProp to sth like 999 instead of 0 in line 899? Ditto in a few other places below where we removed the code to update the last modified time.